### PR TITLE
Extract analyze_unrendered from analyze

### PR DESCRIPTION
**Pull request by zurk**
Use analyze_unrendered in  smoke dataset evaluation.

As was discussed we somehow should separate logic from comment creation.

### CODE REVIEW

File: lookout/style/format/analyzer.py [hunk 1]
```diff
@@ -313,20 +333,21 @@ def _generate_file_comments(
...
-                vnodes=fix_vnodes,            # corrected VirtualNode-s
-                fixed_labels=fix_ys,          # predicted labels for those nodes
+                vnodes=fix_vnodes,            # VirtualNode-s of the line where changes were
+                                              # introduced.
+                winners=fix_rules,            # winner rule indices for the vnodes
+                fixed_labels=fix_ys,          # new predicted labels for those nodes
                 confidence=confidence,        # overall confidence in the prediction, 0-100
-                feature_extractor=fe,         # FeatureExtractor involved.
+                feature_extractor=fe,         # FeatureExtractor involved
+                new_vnodes=new_vnodes,        # fixed vnodes via CodeGenerator for all file
```
Comment (zurk) [inline lookout/style/format/analyzer.py]: We agreed that `new_vnodes` covers `fix_vnodes` and `fix_ys`, but it is not so easy to exclude it and requires more code changes, so I keep it as is for now. 

I can do it now if you want.

File: lookout/style/format/analyzer.py [hunk 2]
```diff
@@ -29,6 +29,20 @@
...
 
+FixData = NamedTuple("FixData", (
+    ("error", str),                           # error message
+    ("language", str),                        # programming language of the code
+    ("line_number", int),                     # line number for the comment
+    ("head_file", File),                      # file from head revision
+    ("base_file", File),                      # file from base revision
+    ("suggested_code", str),                  # code line predicted by our model
+    ("vnodes", List[VirtualNode]),            # VirtualNode-s which correspond to this line
+    ("fixes", List[Tuple[int, int]]),         # changed vnode indices (in `vnodes`)
```
Comment (zurk) [inline lookout/style/format/analyzer.py]: @vmarkovtsev, Let's convert `Tuple[int, int]` to another `NamedTuple` ?

Comment (vmarkovtsev) [reply lookout/style/format/analyzer.py]: I have just understood that we don't need vnode indices at all: we can scan through each vnode in `vnodes` and `getattr(vnode, "y_old") != vnode.y`

Comment (vmarkovtsev) [reply lookout/style/format/analyzer.py]: So `fixes` can be renamed to `winner_rules`

Comment (zurk) [reply lookout/style/format/analyzer.py]: also, we should check line number, but yes, it is possible. just requires linear time.

File: lookout/style/format/analyzer.py [hunk 3]
```diff
@@ -296,37 +321,27 @@ def _generate_file_comments(
...
+            vnodes_changed = [vnode for vnode in new_line_vnodes if
+                              hasattr(vnode, "y_old") and vnode.y_old != vnode.y]
+            yield FixData(
+                error="",                      # success
+                language=language,             # programming language of the code
+                line_number=line_number,       # line number for the comment
+                head_file=file,                # file from head revision
+                base_file=base_file,           # file from base revision
+                suggested_code=new_code_line,  # code line suggested by our model
+                all_vnodes=new_line_vnodes,    # VirtualNode-s which construct the line
```
Comment (vmarkovtsev) [inline lookout/style/format/analyzer.py]: all_vnodes correspond to the whole file
vnodes correspond to the whole line

so

```python
all_vnodes=new_vnodes
vnodes=new_line_vnodes
```

**Comment by zurk** [comment]
Now there is a problem with logging: https://github.com/src-d/lookout-sdk-ml/pull/13

**Comment by vmarkovtsev** [comment]
@zurk Logging fix was merged and released, please update the version

**Comment by zurk** [comment]
So we have the same error because of this bug in `lookout-sdk` https://github.com/src-d/lookout/issues/374

@vmarkovtsev to make Travis happy and make a release I can turn off `uast_break_check` in `evaluate_smoke` code for now. This check produses logs so heavily.

**Pull request merged**